### PR TITLE
Split compressed big files

### PR DESCRIPTION
**Pull request by douglaz**

### CODE REVIEW

**Comment by fparisotto** [comment]
LGTM

**Comment by zago** [comment]
LGTM

File: src/main/scala/ignition/core/utils/AutoCloseableIterator.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  }
+}
+
+trait AutoCloseableIterator[T] extends Iterator[T] with AutoCloseable {
+  // Naive functions should be implemented by the user as in a standard Iterator/AutoCloseable
+  def naiveHasNext(): Boolean
+  def naiveNext(): T
+  def naiveClose(): Unit
+
+  var closed = false
```
Comment (sisso) [inline src/main/scala/ignition/core/utils/AutoCloseableIterator.scala]: public variable?

**Comment by sisso** [comment]
LGTM

File: src/main/scala/ignition/core/utils/AutoCloseableIterator.scala (hunk 1 above)
Comment (douglaz) [inline src/main/scala/ignition/core/utils/AutoCloseableIterator.scala]: just moving code =-D

**Pull request merged**